extensions/community/OneSignal.json

```diff
@@ -0,0 +1,221 @@
+    "network"
+  ],
+  "authorIds": [
+    "jy7FXnGX0ZZcWfrAI9YuQaeIphi1"
+  ],
+  "dependencies": [
+    {
+      "exportName": "onesignal-cordova-plugin",
+      "name": "onesignal-cordova-plugin",
+      "type": "cordova",
+      "version": "3.3.1"
+    }
+  ],
+  "eventsFunctions": [
+    {
+      "description": "Initialize OneSignal.",
+      "fullName": "Initialize OneSignal",
+      "functionType": "Action",
+      "name": "Initialize",
+      "sentence": "Initialize OneSignal",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "const appid = eventsFunctionContext.getArgument(\"appid\");",
+            "const prompt = eventsFunctionContext.getArgument(\"prompt\");",
+            "const permission = eventsFunctionContext.getArgument(\"permission\");",
+            "",
+            "document.addEventListener('deviceready', OneSignalInit, false);",
+            "function OneSignalInit() {",
+            "    // NOTE: Update the init value below with your OneSignal AppId.",
+            "    window.plugins.OneSignal.init(appid);",
+            "    window.plugins.OneSignal.Notifications.setNotificationOpenedHandler(function(opened) {",
+            "        let notificationData = JSON.stringify(opened);",
+            "        console.log('notificationOpenedCallback: ' + notificationData);",
+            "    });",
+            "",
+            "    if (prompt == true) {",
+            "        window.plugins.OneSignal.Notifications.requestPermission(function(accepted) {",
+            "        console.log(\"User accepted notifications: \" + accepted);",
+            "        permission = accepted",
+            "    });",
+            "    }",
+            "    ",
+            "}"
+          ],
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [
+        {
+          "description": "Your onesignal app id",
+          "name": "appid",
+          "type": "string"
+        },
+        {
+          "defaultValue": "yes",
+          "description": "Prompts the user for notification permissions",
+          "name": "prompt",
+          "optional": true,
+          "type": "yesorno"
+        },
+        {
+          "description": "Variable to store notification permission status",
+          "name": "permission",
+          "type": "scenevar"
+        }
+      ],
+      "objectGroups": []
+    },
+    {
+      "description": "Capture Email.",
+      "fullName": "Capture Email",
+      "functionType": "Action",
+      "name": "UserEmail",
+      "sentence": "Capture Email _PARAM1_",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "const email = eventsFunctionContext.getArgument(\"email\");",
+            "",
+            "document.addEventListener('deviceready', OneSignalUserEmail, false);",
```

Comment:
🤔 Have you tested this? I am pretty sure this event only fires once when cordova finishes initializing, it probably won't fire if you don't call this at the beginning of the first scene... Also, every call leaks the callback function you create forever by never unregistering the event listener! 

(Same goes for all other functions in the extension!)